crates/polars-core/src/chunked_array/comparison/mod.rs

```diff
@@ -680,59 +680,90 @@ impl ChunkCompare<&str> for Utf8Chunked {
     }
 }
 
-impl ChunkCompare<&ListChunked> for ListChunked {
-    type Item = BooleanChunked;
-    fn equal(&self, rhs: &ListChunked) -> BooleanChunked {
-        // SAFETY: unstable series never lives longer than the iterator.
-        unsafe {
-            self.amortized_iter()
+#[doc(hidden)]
+fn _list_comparison_helper<F>(lhs: &ListChunked, rhs: &ListChunked, op: F) -> BooleanChunked
+where
+    F: Fn(Option<&Series>, Option<&Series>) -> Option<bool>,
+{
+    match (lhs.len(), rhs.len()) {
+        (_, 1) => {
+            // Safety: values within iterator do not outlive the iterator itself
+            let mut iter = unsafe { rhs.amortized_iter() };
```

Comment:
This could be handled with `.get(0).with_name("")`. It saves us an `unsafe` tag.

No strong opinion here, but I favor keeping the tag count lower.